src/routes/group/chat/[id]/+page.svelte

```diff
@@ -73,94 +108,120 @@
-										</svelte:fragment>
-									</ChatMessage>
-								{:else if message.type === 'data'}
-									<WakuObject {message} users={chat.users} />
-								{/if}
-							{/each}
-						</div>
-					</div>
-				</Container>
-			</div>
-			<div class="chat-input-wrapper">
-				<Container>
-					<div class="chat-input">
-						<Dropdown up left>
-							<!-- TODO: make button "active" while dropdown is open -->
-							<Button variant="icon" slot="button">
-								<Add />
-							</Button>
-							<DropdownItem disabled onClick={() => console.log('Pic from Cam')}
-								>Pic from Cam</DropdownItem
-							>
-							<DropdownItem disabled onClick={() => console.log('Pic from Lib')}
-								>Pic from Lib</DropdownItem
-							>
-							<DropdownItem onClick={() => goto(ROUTES.OBJECTS($page.params.id))}
-								>Waku Object</DropdownItem
-							>
-						</Dropdown>
-						<Textarea
-							placeholder="Message"
-							bind:value={text}
-							on:keypress={(e) => {
-								// When enter is pressed without modifier keys, send the message
-								if (e.key === 'Enter' && !(e.shiftKey || e.ctrlKey || e.altKey)) {
-									sendMessage(wallet)
-									e.preventDefault()
-								}
-								// When shift+enter is pressed, add a newline
-								else if (e.key === 'Enter' && (e.altKey || e.ctrlKey)) {
-									text += '\n'
-									e.preventDefault()
-								}
-							}}
-						/>
-						{#if text.length > 0}
-							<Button variant="strong" disabled={loading} on:click={() => sendMessage(wallet)}>
-								<ArrowUp />
+			<svelte:fragment slot="header">
+				<Header>
+					<Button variant="icon" slot="left" on:click={() => goto(ROUTES.HOME)}>
+						<ChevronLeft />
+					</Button>
+					<svelte:fragment slot="chat">
+						<Avatar picture={chat?.avatar ?? ''} />
+						{chat?.name}
+						<Events />
+					</svelte:fragment>
+					<svelte:fragment slot="right">
+						{#if chat.joined}
+							<Button variant="icon" on:click={() => goto(ROUTES.GROUP_EDIT($page.params.id))}>
+								<Events />
 							</Button>
 						{/if}
-					</div>
+					</svelte:fragment>
+				</Header>
+			</svelte:fragment>
+			{#if !chat.joined}
+				<Container justify="center" alignItems="center" gap={0} padX={24}>
+					<Avatar picture={chat?.avatar ?? ''} size={140} />
+					<Spacer />
+					<p class="text-lg text-bold text-center">Join "{chat?.name}"?</p>
+					<Spacer height={12} />
+					<p class="text-lg text-center">
+						{inviter?.name} invited you to join the "{chat?.name}" group chat.
+					</p>
+					<Spacer />
+					<Container direction="row" justify="center" gap={12} alignItems="center" padY={0}>
+						<Button align="center" variant="strong" on:click={() => join()}>
+							<Checkmark />
+							Join group
+						</Button>
+						<!-- THIS BUTTON WILL BE UN-COMMENTED ONCE THE REMOVE CHAT FUNCTIONALITY IS IMPLEMENTED -->
+						<!-- <Button align="left" on:click={() => decline()}>
+							<Close />
+							Decline
+						</Button> -->
```

Comment:
I see the `decline` function above, what is missing? Can I help?

Reply:
@vojtechsimetka 
@agazso and I worked on it together but there are still some bugs. When the person declines the invitation, they still appear as a member of the group chat and they can still see the chat on their chats list. 
I commented out the button because it's not working and we discussed merging my branch before his (because he worked on some of the same files), but it looks like he ended up merging so I'll turn this back into a draft continue working on it.
If you have availability, I'll gladly accept the help 😊